packages/yoroi-ergo-connector/src/inject.js

```diff
@@ -323,6 +325,19 @@ const ergo = Object.freeze(new ErgoAPI());
 const API_INTERNAL_ERROR = -2;
 const API_REFUSED = -3;
 
+function checkInjectionInDocument() {
+    const el = document.getElementById(INJECTED_TYPE_TAG_ID);
+    return el ? el.value : 'nothing';
+}
+
+function markInjectionInDocument(container) {
+    const inp = document.createElement('input');
+    inp.setAttribute('type', 'hidden');
+    inp.setAttribute('id', INJECTED_TYPE_TAG_ID);
+    inp.setAttribute('value', YOROI_TYPE);
+    container.appendChild(inp);
+}
+
```

Comment:
Why we have to do this?

Reply:
Because the content_script itself does not share the same `window` object, as the code that we are injecting into the DOM. So the DOM itself is the only shared space that allows us to store some information that can be detected by multiple content_scripts executed on the same page.